halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/security/LdapValidator.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2017 Target, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.config.validate.v1.security;
+
+import com.netflix.spinnaker.halyard.config.model.v1.node.Validator;
+import com.netflix.spinnaker.halyard.config.model.v1.security.Ldap;
+import com.netflix.spinnaker.halyard.config.problem.v1.ConfigProblemSetBuilder;
+import com.netflix.spinnaker.halyard.core.problem.v1.Problem;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+@Component
+public class LdapValidator extends Validator<Ldap> {
+
+  @Override
+  public void validate(ConfigProblemSetBuilder p, Ldap ldap) {
+    if (!ldap.isEnabled()) {
+      return;
+    }
+
+    if (StringUtils.isEmpty(ldap.getUrl())) {
```

Comment:
Ok, I can validate the url by using LdapURL, but the build throws a bunch of warnings about the class being proprietary.
`23: warning: LdapURL is internal proprietary API and may be removed in a future release`
`import com.sun.jndi.ldap.LdapURL;`
Their code is available [here](https://github.com/frohoff/jdk8u-dev-jdk/blob/master/src/share/classes/com/sun/jndi/ldap/LdapURL.java)

Is it ok to use this class for validation purposes?